6.0.0 MVU upgrade docs update (#946)
sourcegraph/docs: docs/admin/postgres.mdx

- `codeintel-db`: provides support for lsif data and part of the code-intelligence
- `codeinsights-db`: provides support for code insights data

## Requirements

### Version requirements

> We support Postgres 16 and above. Older versions of Sourcegraph supported 12 see our [Postgres 12 deprecation notice](/admin/postgres12_end_of_life_notice) for more details. Starting in Sourcegraph 6.0.0, sourcegraph services will no longer connect to a PostgreSQL 12 or lower database. Instead displaying an error message in container logs for services attempting to connect to a PostgreSQL database below version 16.

Reviewer: ```suggestion
We support Postgres 16 and above. Older versions of Sourcegraph supported 12 see our [Postgres 12 deprecation notice](/admin/postgres12_end_of_life_notice) for more details. Starting in Sourcegraph 6.0.0, sourcegraph services will no longer connect to a PostgreSQL 12 or lower database. Instead, it will display an error message in container logs for services attempting to connect to a PostgreSQL database below version 16.
```

vector
```

## Configuring PostgreSQL

Sourcegraph databases come preconfigured, however admins may make custom configurations as needed. For more information see our [configuration docs](/admin/config/postgres-conf).

## Upgrading PostgreSQL

### Upgrading Built-in PostgreSQL

In Sourcegraph 5.10.0, we've upgraded our builtin database images to PostgreSQL 16.

Database Image Updates:
- `postgres-12` is now `postgresql-16`
- `codeintel-db` is now `postgresql-16`
- `codeinsights-db` is now `postgresql-16-codeinsights`

> These images contain an entryscript which will detect and upgrade Postgres instances at version 12 (our previous images Postgres version) to Postgres 16.

Reviewer 2: ```suggestion
These images contain an entry script that will detect and upgrade Postgres instances from version 12 (the version in our previous images) to Postgres 16.
```

**For instance specfic instructions on how to upgrade a builtin Postgres database via the image entrypoint script, see [our instance specific operational instructions](/admin/how-to/upgrade-postgres-12-16-builtin-dbs).**

We have the following advisements to admins:
- Container orchestration management systems (e.g. Kubernetes) may restart containers at any time, **it is recommended that you take a backup of your database before starting the upgrade process.**
- Review your deployment types [upgrade notes](/admin/updates#upgrades-index) before upgrading. The PostgreSQL upgrade process mutates the database schema in a way that may Sourcegraph `v5.10.0` and `v5.11.0` may not recognize. [Learn more about pg12 to pg16 schema drift here](/admin/how-to/postgres_12_to_16_drift).

For additional assistance with PostgreSQL upgrades, please contact support@sourcegraph.com.

### Upgrade entrypoint script options

The `PG_UPGRADE_EXTRA_ARGS` environment variable allows you to customize the `pg_upgrade` command in `postgresql-16` and `postgresql-16-codeinsights` deployments.

- **Quick upgrade option:**

> Setting `PG_UPGRADE_EXTRA_ARGS=--link` creates hard links between old and new data directories instead of copying data. While faster and space-efficient, this method risks data corruption if containers are interrupted during upgrade. Best used when you have reliable container scheduling and solid backups.

Reviewer 2: ```suggestion
Setting `PG_UPGRADE_EXTRA_ARGS=--link` creates hard links between old and new data directories instead of copying data. While faster and space-efficient, this method risks data corruption if containers are interrupted during an upgrade. It is best used when you have reliable container scheduling and solid backups.
```
